### PR TITLE
feat: Add UserAction to GenericEvents

### PR DESCRIPTION
**Pull request by cwli24**
Select user interactive events will be captured in this release. This is an extension of the GenericEvents feature meant to drive UI experience.
---

### Overview

Add `UserAction` event type going to `ins` endpoint. The list of select events are auto captured for DOM elements.

### Related Issue(s)

https://new-relic.atlassian.net/browse/NR-304358

### Testing

Added

### CODE REVIEW

**Review by metal-messiah — APPROVED** [review]
LGTM, when tests pass this is G2G

File: src/features/generic_events/instrument/index.js [hunk 1]
```diff
@@ -2,19 +2,34 @@
...
     super(agentIdentifier, aggregator, FEATURE_NAME, auto)
+    const agentInit = getConfiguration(this.agentIdentifier)
     const genericEventSourceConfigs = [
-      getConfigurationValue(this.agentIdentifier, 'page_action.enabled')
+      agentInit.page_action.enabled,
+      agentInit.user_actions.enabled
       // other future generic event source configs to go here, like M&Ms, PageResouce, etc.
     ]
+
+    if (agentInit.user_actions.enabled) {
```
Comment (metal-messiah) [inline src/features/generic_events/instrument/index.js:23]: I think this just needs to check `isBrowserScope` in this statement for safety

**Comment by cwli24** [comment]
This is waiting for https://source.datanerd.us/browser/dirac_browser_consumer/pull/430

File: tools/testing-server/constants.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ module.exports.defaultInitBlock = {
   metrics: { enabled, autoStart },
   obfuscate: undefined,
   page_action: { enabled },
+  user_actions: { enabled: false },
```
Comment (metal-messiah) [inline tools/testing-server/constants.js:62]: Does this still need to happen with the adjustments that have been made to the page actions and user actions tests?  I think we'd be better off if the default state of the agent matched the default state of the testing server configs, if feasible

Comment (cwli24) [reply tools/testing-server/constants.js:62]: I didn't fix every insightCapture case that's why. I'm just going to throw in a separate story into backlog for that.

Comment (metal-messiah) [reply tools/testing-server/constants.js:62]: That works for me

**Review by metal-messiah — APPROVED** [review]

**Comment by metal-messiah** [comment]
https://github.com/newrelic/newrelic-browser-agent/actions/runs/10966174564

**Pull request merged**